src/runtime/pkg/containerd-shim-v2/create.go

```diff
@@ -110,21 +111,35 @@ func withCDI(annotations map[string]string, cdiSpecDirs []string, spec *specs.Sp
 
 func copyLayersToMounts(rootFs *vc.RootFs, spec *specs.Spec) error {
 	for _, o := range rootFs.Options {
-		if !strings.HasPrefix(o, annotations.FileSystemLayer) {
-			continue
-		}
+		if strings.HasPrefix(o, annotations.FileSystemLayer) {
 
-		fields := strings.Split(o[len(annotations.FileSystemLayer):], ",")
-		if len(fields) < 2 {
-			return fmt.Errorf("Missing fields in rootfs layer: %q", o)
-		}
+			fields := strings.Split(o[len(annotations.FileSystemLayer):], ",")
+			if len(fields) < 2 {
+				return fmt.Errorf("Missing fields in rootfs layer: %q", o)
+			}
 
-		spec.Mounts = append(spec.Mounts, specs.Mount{
-			Destination: "/run/kata-containers/sandbox/layers/" + filepath.Base(fields[0]),
-			Type:        fields[1],
-			Source:      fields[0],
-			Options:     fields[2:],
-		})
+			spec.Mounts = append(spec.Mounts, specs.Mount{
+				Destination: "/run/kata-containers/sandbox/layers/" + filepath.Base(fields[0]),
+				Type:        fields[1],
+				Source:      fields[0],
+				Options:     fields[2:],
+			})
+		} else if strings.HasPrefix(o, "extraoption=") {
```

Comment:
How about using `io.katacontainers.volume` instead of `extraoption`?

Reply:
the Option in rootFs is passed from nydus-snapshotter, which is encoded by "extraoption={data}". So here we can only read data by matching the “extraoption=” prefix.

Reply:
We are going to enhance nydus-snapshotter to use `io.katacontainers.volume` instead of `extraoption`.